### PR TITLE
Fix csv transforms in covid pipeline

### PR DESCRIPTION
**Pull request by mpbrown**
# BACKEND PULL REQUEST

## Related Issue

- The covid csv pipeline needs to perform some preprocess transforms like SNOMED code mapping and datetime handling on the uploaded data, but the current implementation is unable to handle escaped commas in the data.
- #6025 

## Changes Proposed

- When preprocessing on the csv pipeline, use the same mapping iterator used in the FHIR bundle pipeline to safely parse the csv, perform necessary changes, and rewrite to a byte array.

## Testing

- Submit test files to the csv uploader to see if we are missing any edge cases
- Deploying on dev7

### CODE REVIEW

File: backend/src/main/java/gov/cdc/usds/simplereport/service/TestResultUploadService.java [hunk 1]
```diff
@@ -163,109 +171,115 @@ public TestResultUpload processResultCSV(InputStream csvStream) {
...
+
+    var specimenCollectionDate =
+        StringUtils.isNotBlank(row.get(SPECIMEN_COLLECTION_DATE_COLUMN_NAME))
             ? convertToZonedDateTime(
-                    specimenCollectionDate, resultsUploaderCachingService, providerAddr)
-                .toOffsetDateTime()
-                .toString()
+                row.get(SPECIMEN_COLLECTION_DATE_COLUMN_NAME),
+                resultsUploaderCachingService,
+                providerAddr)
```
Comment (zdeveloper) [inline backend/src/main/java/gov/cdc/usds/simplereport/service/TestResultUploadService.java:245]: Does this need to be changed to the testingLabAddr

File: backend/src/main/java/gov/cdc/usds/simplereport/service/TestResultUploadService.java [hunk 2]
```diff
@@ -163,109 +171,115 @@ public TestResultUpload processResultCSV(InputStream csvStream) {
...
-            .toOffsetDateTime()
-            .toString();
-
-    specimenCollectionDate =
-        StringUtils.isNotBlank(specimenCollectionDate)
+  private Map<String, String> transformCsvRow(Map<String, String> row) {
+    var updatedSpecimenType =
+        modifyRowSpecimenNameToSNOMED(row.get(SPECIMEN_TYPE_COLUMN_NAME).toLowerCase());
+
+    var testingLabAddr =
```
Comment (zdeveloper) [inline backend/src/main/java/gov/cdc/usds/simplereport/service/TestResultUploadService.java:215]: rename to testingLabAddress

File: backend/src/main/java/gov/cdc/usds/simplereport/service/TestResultUploadService.java [hunk 3]
```diff
@@ -163,109 +171,115 @@ public TestResultUpload processResultCSV(InputStream csvStream) {
...
+        new StreetAddress(
+            row.get("testing_lab_street"),
+            row.get("testing_lab_street2"),
+            row.get("testing_lab_city"),
+            row.get("testing_lab_state"),
+            row.get("testing_lab_zip_code"),
+            null);
+    var providerAddr =
```
Comment (zdeveloper) [inline backend/src/main/java/gov/cdc/usds/simplereport/service/TestResultUploadService.java:223]: rename to providerAddress

**Review by zdeveloper — COMMENTED** [review]
LGTM, with some questions, will test in dev7 then approve

File: backend/src/main/java/gov/cdc/usds/simplereport/service/TestResultUploadService.java (hunk 1 above)
Comment (mpbrown) [reply backend/src/main/java/gov/cdc/usds/simplereport/service/TestResultUploadService.java]: From #5987, a blank `specimen_collection_date` field is populated by `order_test_date` so based on that I'm assuming we should use the ordering provider address for both of those fields

File: backend/src/main/java/gov/cdc/usds/simplereport/service/TestResultUploadService.java [hunk 4]
```diff
@@ -185,30 +182,37 @@ private byte[] transformCsvContent(byte[] content) {
...
 
-    return csvContent.toString().getBytes(StandardCharsets.UTF_8);
+    return csvContent.getBytes(StandardCharsets.UTF_8);
   }
 
   private Map<String, String> transformCsvRow(Map<String, String> row) {
+
+    if (!"P".equals(processingModeCodeValue)
+        && !row.containsKey(PROCESSING_MODE_CODE_COLUMN_NAME)) {
+      row.put(PROCESSING_MODE_CODE_COLUMN_NAME, processingModeCodeValue);
```
Comment (mpbrown) [inline backend/src/main/java/gov/cdc/usds/simplereport/service/TestResultUploadService.java:209]: I like that you moved this code into the transform method so that the csv changes all happen in one place!

**Review by zdeveloper — APPROVED** [review]
LGTM!

File: backend/src/test/java/gov/cdc/usds/simplereport/service/TestResultUploadServiceTest.java [hunk 5]
```diff
@@ -385,8 +386,8 @@ void uploadService_getUploadSubmission_processingModeCode_doesntOverrideFileValu
     // THEN
     verify(dataHubMock).uploadCSV(fileContentCaptor.capture());
     String[] rows = new String(fileContentCaptor.getValue(), StandardCharsets.UTF_8).split("\n");
-    assertThat(rows[0]).endsWith(",processing_mode_code");
-    assertThat(rows[1]).endsWith(",D");
+    assertThat(rows[0]).endsWith(",\"processing_mode_code\"");
+    assertThat(rows[1]).endsWith(",\"D\"");
```
Comment (nathancrtr) [inline backend/src/test/java/gov/cdc/usds/simplereport/service/TestResultUploadServiceTest.java:390]: How come this changed? Is this the effect of the `ALWAYS_QUOTE_STRINGS` feature for the `CsvMapper` I'm guessing?

Comment (zdeveloper) [reply backend/src/test/java/gov/cdc/usds/simplereport/service/TestResultUploadServiceTest.java:390]: yes, that's a correct guess

Comment (zdeveloper) [reply backend/src/test/java/gov/cdc/usds/simplereport/service/TestResultUploadServiceTest.java:390]: It was adding quotes to strings, but not always, and I added the `ALWAYS_QUOTE_STRINGS` setting to make it consistent

**Review by nathancrtr — APPROVED** [review]
This looks great! Thanks for the thorough tests and the bit of refactoring as well, the service code is much more readable in its new form.
